src/itype.h

```diff
@@ -1046,10 +1046,6 @@ struct itype {
         // A list of conditional names, in order of ascending priority.
         std::vector<conditional_name> conditional_names;
 
-        // Since the material list was converted to a map, keep track of the material insert order
-        // Do not use this for materials. Use the materials map above.
-        std::vector<material_id> mats_ordered;
```

Comment:
IIRC I did this to preserve some legacy behaviour that expects materials to be listed in the order they were defined. You could totally get around this by using a paired vector instead of a map to preserve the inserted order:

```diff
diff --git a/src/itype.h b/src/itype.h
index f41e2413e9..e68bb4b647 100644
--- a/src/itype.h
+++ b/src/itype.h
@@ -1099,7 +1099,7 @@ struct itype {
         // First -> the material
         // Second -> the portion of item covered by the material (portion / total portions)
         // MATERIALS WORK IN PROGRESS.
-        std::map<material_id, int> materials;
+        std::vector<std::pair<material_id, int>> materials;
 
         /** Actions an instance can perform (if any) indexed by action type */
         std::map<std::string, use_function> use_methods;
```

This way you wouldn't need `first_mat`. (Not sure why I didn't do that in the first place)

Reply:
hey thanks for responding to this I was going to bug you about this today if tests started yelling. I imagine at some point the legacy stuff should just go away.  Currently though I'm just doing this to cut down on the amount of duplicated work I have to do since copying every material twice is more work so I thought I could at least shave this down to a single material. I'm gonna make it more clear that it's the "default_mat" instead of the more confusing "first_mat".

I think the benefits of a map do outweigh the ordering of pairs like you said.